File: spring-cloud-integrated-demo/event-lib/src/main/java/com/github/jccode/springcloud/integrateddemo/common/event/model/Event.java
package com.github.jccode.springcloud.integrateddemo.common.event.model;

import java.io.Serializable;

public class Event implements Serializable {

    /**
     * 事件ID
     */
    private String uuid;

    /**
     * 事件的名称
     */
    private String type;

    /**
     * 数据
     */
    private String payload;

    public Event() {
    }

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getPayload() {
        return payload;
    }

    public void setPayload(String payload) {
        this.payload = payload;
    }
}

File: spring-boot-kafka-demo/kafka-json-demo2/src/main/scala/com/github/jccode/springbootkafkademo/jsondemo22/event/Event.scala
package com.github.jccode.springbootkafkademo.jsondemo22.event

import java.util.UUID

import scala.beans.BeanProperty


class Event[T](@BeanProperty val uuid: String,
               @BeanProperty val eventType: String,
               @BeanProperty val payload: T) extends Serializable {

  def this(eventType: String, payload: T) = this(UUID.randomUUID.toString, eventType, payload)

  override def toString = s"Event($uuid, $eventType, $payload)"
}

object Event {

  def apply[T](uuid: String, eventType: String, payload: T): Event[T] = new Event(uuid, eventType, payload)

  def apply[T](eventType: String, payload: T): Event[T] = new Event(eventType, payload)

}
